app/cdap/components/DirectiveInput/Components/AutoComplete/index.tsx

```diff
@@ -0,0 +1,236 @@
+  isDirectiveSelected,
+  columnNamesList,
+  onRowClick,
+  getDirectiveUsage,
+  onColumnSelected,
+  directiveInput,
+}: IAutoCompleteProps) {
+  const [activeResults, setActiveResults] = useState<IUsageDirectives[]>([]);
+  const [input, setInput] = useState<string>('');
+  const [activeSelectionIndex, setActiveSelectionIndex] = useState<number | null>(null);
+  const eventEmitter = ee(ee);
+  const [fuse, setFuse] = useState(null);
+
+  const getUsage = () => {
+    if (!isDirectiveSelected) {
+      MyDataPrepApi.getUsage({ context: NamespaceStore.getState().selectedNamespace }).subscribe(
+        (res) => {
+          setDirectivesList(res.values);
+          setFuse(new Fuse(res.values, { ...defaultFuseOptions }));
+        }
+      );
+    } else {
+      setFuse(new Fuse(columnNamesList, { ...defaultFuseOptions, keys: ['label'] }));
+    }
+  };
+
+  useEffect(() => {
+    getUsage();
+  }, [isDirectiveSelected]);
+
+  useEffect(() => {
+    eventEmitter.on(globalEvents.DIRECTIVEUPLOAD, getUsage);
+    const directiveInput = document.getElementById('directive-input-search');
+    const mousetrap = new Mousetrap(directiveInput);
+    mousetrap.bind('up', handleUpArrow);
+    mousetrap.bind('down', handleDownArrow);
+    mousetrap.bind('enter', handleEnterKey);
+    mousetrap.bind('tab', handleTabKey);
+
+    // returned function will be called on component unmount
+    return () => {
+      mousetrap.unbind('up');
+      mousetrap.unbind('down');
+      mousetrap.unbind('enter');
+      mousetrap.unbind('tab');
+      eventEmitter.off(globalEvents.DIRECTIVEUPLOAD, getUsage);
+    };
+  });
+  const handleUpArrow = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (activeSelectionIndex !== 0) {
+      setActiveSelectionIndex(activeSelectionIndex - 1);
+    }
+  };
+
+  const handleDownArrow = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (activeSelectionIndex !== activeResults.length - 1) {
+      setActiveSelectionIndex(activeSelectionIndex + 1);
+    }
+  };
+
+  const handleEnterKey = () => {
+    if (input.length > 0) {
+      if (activeResults[activeSelectionIndex]) {
+        handleRowClick(activeResults[activeSelectionIndex]);
+      } else {
+        onRowClick({
+          target: { value: `${input}` },
+        });
+      }
+    }
+  };
+
+  const handleTabKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (input.length === 0 || input.split(' ').length !== 1) {
+      return;
+    }
+    event.preventDefault();
+    handleEnterKey();
+  };
+
+  useEffect(() => {
+    if (isDirectiveSelected) {
+      setFuse(new Fuse(columnNamesList, { ...defaultFuseOptions, keys: ['label'] }));
+    }
+    searchMatch(directiveInput);
+    setInput(directiveInput);
+  }, [directiveInput]);
+
+  const searchMatch = (query: string) => {
+    let results = [];
+    const input: string = query;
+    const spaceIndex: number = input.indexOf(' ');
```

Comment:
why are we giving space here?

Reply:
As soon as directive is entered, we need column list to appear hence we are checking if space is present in it, we replace above line of code to  ------   const spaceIndex: boolean = searchString.includes(' ');